toss/trajectory_tools.py

```diff
@@ -34,32 +39,87 @@ def compute_trajectory(x: np.ndarray, args, func: Callable) -> Union[np.ndarray,
+                time_of_maneuver (float): Time for adding impulsive maneuver [seconds].
+                delta_v (np.ndarray): Array containing the cartesian componants of the impulsive maneuver.
         func (Callable): A function handle for the state update equation required for integration.
 
     Returns:
         trajectory_info (np.ndarray): Numpy array containing information on position and velocity at every time step (columnwise).
         squared_altitudes (float): Sum of squared altitudes above origin for every position
         collision_penalty (bool): Penalty value given for the event of a collision with the celestial body.
     """
-    
-    # Compute trajectory by numerical integration
-    trajectory, trajectory_info = integrate(func, x, args)
+    # Convert osculating orbital elements to cartesian for integration
+    r, v = pk.par2ic(E=x[0:6], mu=args.body.mu)
+    x_cartesian = np.array(r+v)
+
+    # Setup time intervals with/without maneuvers
+    if args.problem.number_of_maneuvers == 0:
+        time_list = [args.problem.start_time, args.problem.final_time]
+    else:
+        # Add state variables related to the impulsive maneuver in args
+        time_list = np.empty((args.problem.number_of_maneuvers+2), dtype=int)
+        delta_v_array = np.empty((3, args.problem.number_of_maneuvers), dtype=np.float64)
+        m = 0
+        time_list[0] = args.problem.start_time
+        time_list[-1] = args.problem.final_time
+        for n in range(6, len(x)-3, 4):
+            time_list[m+1] = int(x[n])
+            delta_v_array[:,m] = [x[n+1], x[n+2], x[n+3]]
+            m += 1
+
+        # Rearrange maneuvers in increasing order of time of execution
+        maneuver_times = time_list[1:-1]
+        correct_order_of_maneuvers = np.argsort(maneuver_times)
+        time_list[1:-1] = maneuver_times[correct_order_of_maneuvers]
+        delta_v_array = delta_v_array[:,correct_order_of_maneuvers]
+
+    # Integrate trajectory for each subinterval
+    trajectory_info = 0
+    for i in range(0, len(time_list)-1):
+        args.integrator.t0 = time_list[i]
+        args.integrator.tf = time_list[i+1]
+
+        if i > 0:
+            # Start at end position of previous interval, now adding impulsive manuever
+            x_cartesian = trajectory_info[0:6,-1]
+            x_cartesian[3:6] += delta_v_array[:,i-1]
+
+        # Compute trajectory by numerical integration
+        trajectory, trajectory_memory = integrate_trajectory(func, x_cartesian, args)
+
+        if i == 0:
+            trajectory_info = trajectory_memory
+
+        else:
+            trajectory_info = np.hstack((trajectory_info[:,0:-1] ,trajectory_memory))
+        
+        # Check for potential collisions
+        event_triggers = np.empty((len(trajectory.events), 3), dtype=np.float64)
+        k = 0
+        for event in trajectory.events:
+            event_triggers[k,:] = event.y[0:3]
+            k += 1
```

Comment:
what is k? descriptive names. And why are the events a vec3? :thinking:

Reply:
Fixed.